realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -1025,6 +1030,39 @@ private Table getPrimaryKeyTable() {
         return pkTable;
     }
 
+    /**
+     * Rename the primary key column for the table. This needs to be called manually before inserting data into the table.
+     *
+     * @param newFieldName the name of the field replacing old primary key field name.
+     * @throws {@link IllegalArgumentException} if {@code newFieldName} is an empty string, or exceeds field name length limit.
+     * @throws {@link io.realm.exceptions.RealmException} if a primary key does not exist.
+     */
+    public void renamePrimaryKeyColumn(String newFieldName) {
```

Comment:
Instead of having a separate method for this, why not check for this special case in `renameColumn` ? Right now this is a bit inconsistent with addColumn and removeColumn which do not have a special `*PrimaryKeyColumn` method (nor do they need one IMO)